src/shared/components/results/ResultsData.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useRef, useMemo, FC } from 'react';
+import {
+  DataTableWithLoader,
+  DataListWithLoader,
+  Loader,
+  PageIntro,
+} from 'franklin-sites';
+import { useHistory, useLocation } from 'react-router-dom';
+
+import NoResultsPage from '../error-pages/NoResultsPage';
+import ResultsButtons from './ResultsButtons';
+
+import useNS from '../../hooks/useNS';
+import useNSQuery from '../../hooks/useNSQuery';
+import useUserPreferences from '../../hooks/useUserPreferences';
+import useColumns from '../../hooks/useColumns';
+import usePagination from '../../hooks/usePagination';
+import useItemSelect from '../../hooks/useItemSelect';
+
+import { getIdKeyFor } from '../../utils/getIdKeyForNamespace';
+import { getParamsFromURL } from '../../../uniprotkb/utils/resultsUtils';
+
+import { getEntryPathFor } from '../../../app/config/urls';
+import cardRenderer from '../../config/resultsCardRenderers';
+import infoMappings from '../../config/InfoMappings';
+
+import { Namespace } from '../../types/namespaces';
+import { APIModel } from '../../types/apiModel';
+
+import './styles/warning.scss';
+import './styles/results-data.scss';
+
+export enum ViewMode {
+  TABLE,
+  CARD,
+}
+
+const ResultsData: FC = () => {
+  const namespace = useNS() || Namespace.uniprotkb;
+  const [viewMode] = useUserPreferences<ViewMode>('view-mode', ViewMode.CARD);
+  const history = useHistory();
+  const [columns, updateColumnSort] = useColumns();
+  const [selectedEntries, handleEntrySelection] = useItemSelect();
+  const { search: queryParamFromUrl } = useLocation();
+  const { query, selectedFacets, sortColumn, sortDirection } = getParamsFromURL(
```

Comment:
Could this be done in `ResultsButtons` instead?

Reply:
Good point, I did that then realised it could be moved further down to Download. Updated.